slurm/train.slurm

```diff
@@ -61,6 +56,83 @@ echo "Gradient accumulation steps: $GRAD_ACC_STEPS"
 MASTER_ADDR=$(scontrol show hostnames $SLURM_JOB_NODELIST | head -n 1)
 MASTER_PORT=6000
 
+# Get all hostnames in allocation
+HOSTNAMES=($(scontrol show hostnames $SLURM_JOB_NODELIST))
+NUM_HOSTS=${#HOSTNAMES[@]}
+LAST_NODE=${HOSTNAMES[-1]}
+LAST_NODE_IP=$(getent hosts $LAST_NODE | awk '{ print $1 }')
```

Comment:
This does not work? `LAST_NODE_IP=(scontrol show hostnames $SLURM_JOB_NODELIST | tail -n 1)`